tensorboard/webapp/metrics/views/main_view/card_group_toolbar_container.ts

```diff
@@ -0,0 +1,74 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  SimpleChanges,
+  OnChanges,
+  OnDestroy,
+} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {BehaviorSubject, Observable, of, Subject} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {State} from '../../../app_state';
+import {metricsTagGroupExpansionChanged} from '../../actions';
+import {getMetricsTagGroupExpansionState} from '../../../selectors';
+
+@Component({
+  selector: 'metrics-card-group-toolbar',
+  template: `
+    <metrics-card-group-toolbar-component
+      [numberOfCards]="numberOfCards"
+      [isGroupExpanded]="isGroupExpanded$ | async"
+      [groupName]="groupName"
+      (groupExpansionToggled)="onGroupExpansionToggled()"
+    ></metrics-card-group-toolbar-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CardGroupToolBarContainer {
+  @Input() groupName: string | null = null;
+  @Input() numberOfCards!: number;
+  @Input() isFirstGroup!: boolean;
+  isGroupExpanded$: Observable<boolean> = of(false);
+
+  constructor(private readonly store: Store<State>) {}
+
+  ngOnInit() {
+    if (this.isFirstGroup && this.groupName) {
+      this.store.dispatch(
+        metricsTagGroupExpansionChanged({tagGroup: this.groupName})
+      );
+    }
```

Comment:
Please do not dispatch an action on the Component/DOM mount. This is analogous to a composite action. This is used because we did not have much choice for `coreLoaded` but it is disallowed.